### PR TITLE
modules: openthread: Fix building with picolibc

### PR DESCRIPTION
**Pull request by pdgendt**
The OpenThread module should not explicitly link with libc in case of minimal libc or picolibc.

### CODE REVIEW

**Comment by stephanosio** [comment]
Rationale given here: https://github.com/zephyrproject-rtos/openthread/commit/b7433ff4ef9e0d8348bdc605ff82efb9c356b233#

File: modules/openthread/CMakeLists.txt [hunk 1]
```diff
@@ -433,9 +433,11 @@ target_compile_options(ot-config INTERFACE
 # errno_private.h is generated as part of ${SYSCALL_LIST_H_TARGET} target.
 add_dependencies(ot-config ${SYSCALL_LIST_H_TARGET})
 
-# Make sure C library is linked after OpenThread libraries (to prevent linker
-# errors)
-target_link_libraries(ot-config INTERFACE -lc)
+# Make sure C library, in case of newlib, is linked after OpenThread libraries
+# (to prevent linker errors)
+if(CONFIG_NEWLIB_LIBC)
+  target_link_libraries(ot-config INTERFACE -lc)
```
Comment (stephanosio) [inline modules/openthread/CMakeLists.txt:439]: My question is: why was this necessary in the first place? https://github.com/zephyrproject-rtos/zephyr/pull/48646#issuecomment-1204271028 describes what problem this is trying to fix, but does not explain the root cause.

Also is this still necessary?

Comment (pdgendt) [reply modules/openthread/CMakeLists.txt:439]: It is still required in case of `newlib`, I didn't dig deeper, just verified with `minimal/picolibc/newlib`.

Comment (stephanosio) [reply modules/openthread/CMakeLists.txt:439]: > It is still required in case of `newlib`

Ok, it would be nice if we can eventually get rid of this hack, but that is beyond the scope of this PR.

**Review by stephanosio — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
